clusterloader2/pkg/execpod/exec_pod.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package execpod
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+
+	"k8s.io/perf-tests/clusterloader2/pkg/framework"
+	"k8s.io/perf-tests/clusterloader2/pkg/framework/client"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+)
+
+const (
+	execPodDeploymentNamespace = "cluster-loader"
+	execPodDeploymentName      = "exec-pod"
+	execPodReplicas            = 3
+	execPodSelector            = "pod = exec"
+	execPodPath                = "pkg/execpod/manifest/exec_pod.yaml"
+
+	execPodCheckInterval = 10 * time.Second
+	execPodCheckTimeout  = 2 * time.Minute
+)
+
+// SetUpExecPod creates exec pod.
+func SetUpExecPod(f *framework.Framework) error {
+	klog.Info("Setting up exec pod")
+	mapping := make(map[string]interface{})
+	mapping["Name"] = execPodDeploymentName
+	mapping["Namespace"] = execPodDeploymentNamespace
+	mapping["Replicas"] = execPodReplicas
+	if err := client.CreateNamespace(f.GetClientSets().GetClient(), execPodDeploymentNamespace); err != nil {
+		return fmt.Errorf("namespace %s creation error: %v", execPodDeploymentNamespace, err)
+	}
+	if err := f.ApplyTemplatedManifests(
+		execPodPath,
+		mapping,
+		client.Retry(apierrs.IsNotFound)); err != nil {
+		return fmt.Errorf("pod %s creation error: %v", execPodDeploymentName, err)
+	}
+	ps, err := measurementutil.NewPodStore(f.GetClientSets().GetClient(), execPodDeploymentNamespace, execPodSelector, "")
```

Comment:
Can't we reuse some functionality of WaitForPods measumrent? Maybe we can extract this code somehow?

Reply:
Done. #638